src/map/latent_effect_container.cpp

```diff
@@ -587,6 +587,15 @@ void CLatentEffectContainer::CheckLatentsEquip(uint8 slot)
                 case LATENT_NATION_CONTROL:
                         CheckLatentsZone();
                     break;
+                case LATENT_WEAPON_DRAWN:
+		case LATENT_WEAPON_DRAWN_HP_UNDER:
+		case LATENT_WEAPON_SHEATHED:
+		{
+			ACTIONTYPE action = m_POwner->PBattleAI->GetCurrentAction();
+
+			CheckLatentsWeaponDraw(action == ACTION_ATTACK);

```

Comment:
do m_POwner->animation == ANIMATION_ATTACK here instead (or else you won't get it if you're casting a spell or in the middle of a WS or something)